cns/middlewares/SWIFTv2.go

```diff
@@ -0,0 +1,111 @@
+package middlewares
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/Azure/azure-container-networking/cns"
+	"github.com/Azure/azure-container-networking/cns/configuration"
+	"github.com/Azure/azure-container-networking/cns/types"
+	"github.com/Azure/azure-container-networking/crd/multitenancy/api/v1alpha1"
+	v1 "k8s.io/api/core/v1"
+	k8stypes "k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var ErrMTPNCNotReady = errors.New("mtpnc is not ready")
+
+type SWIFTv2Middleware struct {
+	// TODO: implement
+	// need cached scoped client for pods
+	// need client for MTPNC CRD for x-ref pods
+	cli client.Client
+}
+
+func NewSWIFTv2Middleware(cli client.Client) *SWIFTv2Middleware {
+	return &SWIFTv2Middleware{
+		cli: cli,
+	}
+}
+
+// Return the validator function for the middleware
+func (m *SWIFTv2Middleware) Validator() cns.IPConfigValidator {
+	return m.validateMultitenantIPConfigsRequest
+}
```

Comment:
why do we need this extra layer of indirection? it seems like we could just have a `Validate(req)` method?